src/Discord.Net/Entities/Users/IMember.cs

```diff
@@ -0,0 +1,38 @@
+// todo: meth
+// todo: prop
+// todo: docs
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Wumpus.Entities;
+
+namespace Discord
+{
+    public interface IMember : ISnowflakeEntity, IVoiceState
+    {
+        ValueTask<IUser> GetUserAsync(StateBehavior stateBehavior = default, RequestOptions? options = default);
+        ValueTask<IGuild> GetGuildAsync(StateBehavior stateBehavior = default, RequestOptions? options = default);
+        ValueTask<IReadOnlyCollection<IRole>> GetRoles(StateBehavior stateBehavior = default, RequestOptions? options = default);
+
+        ulong GuildId { get; }
+        IReadOnlyCollection<ulong> RoleIds { get; }
+        IMemberPresence Presence { get; }
+
+        DateTimeOffset? JoinedAt { get; }
+        string Nickname { get; }
```

Comment:
nit: thoughts on a DisplayName extension that does `Nickname ?? Username`, since it's asked for so often

Reply:
Seconded.